status/statistics.py

```diff
@@ -22,19 +23,55 @@ def get_stats_data(db,view,gl=None):
             data[row.key]=row.value
     return data
 
+def clean_application_keys_y(raw_data):
+    # Normalise Application names to approved list
+    # Don't let Denis see this, it will make him cry
+    with open('application_keys.yaml') as f:
+        app_cats=yaml.load(f)
+    data={}
+    for year in raw_data:
+        data[year] = {}
+        for c in raw_data[year]:
+            try:
+                data[year][app_cats[c]] += raw_data[year][c]
+            except KeyError:
+                try:
+                    data[year][app_cats[c]] = raw_data[year][c]
+                except KeyError:
+                    data[year][c] = raw_data[year][c]
```

Comment:
If you want to reduce somelines (or hurt someones head), you could replace `try` and `except` loop with

```
data[year][app_cats.get(c, c)] = data[year].get(app_cats.get(c, c),"") + raw_data[year][c]
```